File: week4/lee/큰수만들기.py
def solution(number, k):
    answer = []

    for i in number:
        while k>0 and answer and answer[-1] < i: # 순서대로 돌아가면서 큰수가 나오면 앞에넣기
            answer.pop()
            k -= 1
        answer.append(i)
   
    return ''.join(answer[:len(answer)-k])

print(solution("332211"	,2))

File: week4/eunbeen/큰수만들기.java
import java.util.*;

class Solution {
    public String solution(String number, int k) {
        if(number.length() == k) return number;
        
        Stack<Character> stack = new Stack();
        var result = new StringBuilder();
        
        for(int i=0; i<number.length(); i++){
            char c = number.charAt(i);
            while (!stack.isEmpty() && k > 0 && stack.peek() < c) {
                stack.pop();
                k--;
            }
            stack.push(c);
        }
        
        while (k-- > 0) {
            stack.pop();
        }

        for (char c : stack) {
            result.append(c);
        }
        
        return result.toString();
    }
}